app/controllers/users_controller.rb

```diff
@@ -247,4 +248,12 @@ def set_email_preferences
       @email_prefs[pref.email_type] = false
     end
   end
+
+  def change_privacy_params(key, value)
```

Comment:
I think you can simply use `change_settings` in `update_privacy_settings` and drop this method here.